desktop/cmp/input/CodeInput.ts

```diff
@@ -240,210 +282,227 @@ class CodeInputModel extends HoistInputModel {
-            run: value => {
-                const {editor} = this;
-                if (editor && editor.getValue() != value) {
-                    // CodeMirror will throw on null value.
-                    editor.setValue(value == null ? '' : value);
+            run: val => {
+                if (this.editor && this.editor.state.doc.toString() !== val) {
+                    this.editor.dispatch({
+                        changes: {from: 0, to: this.editor.state.doc.length, insert: val ?? ''}
+                    });
                 }
             }
         });
 
         this.addReaction({
             track: () => this.componentProps.readonly || this.componentProps.disabled,
-            run: editorReadOnly => {
-                this.editor.setOption('readOnly', editorReadOnly);
+            run: readOnly => {
+                if (this.editor)
+                    this.editor.dispatch({
+                        effects: StateEffect.appendConfig.of(EditorView.editable.of(!readOnly))
+                    });
             }
         });
 
         this.addReaction({
             track: () => this.query,
             run: query => {
-                if (query?.trim()) {
-                    this.findAll();
-                } else {
-                    this.clearSearchResults();
-                }
+                if (query?.trim()) this.findAll();
+                else this.clearSearchResults();
             },
             debounce: 300
         });
     }
 
-    manageCodeEditor = textAreaComp => {
-        if (textAreaComp) {
-            this.editor = this.createCodeEditor(textAreaComp);
-            this.preserveSearchResults();
-        }
-    };
-
-    createCodeEditor(textAreaComp) {
-        const editorSpec = defaultsDeep(this.componentProps.editorProps, this.createDefaults());
-
-        const taDom = findDOMNode(textAreaComp),
-            editor = codemirror.fromTextArea(taDom, editorSpec);
-
-        editor.on('change', this.handleEditorChange);
-        return editor;
-    }
-
-    createDefaults() {
-        const {disabled, readonly, mode, linter, autoFocus} = this.componentProps;
-        let gutters = ['CodeMirror-linenumbers', 'CodeMirror-foldgutter'];
-        if (linter) gutters.push('CodeMirror-lint-markers');
-
-        return {
-            mode,
-            theme: XH.darkTheme ? 'dracula' : 'default',
-            lineWrapping: false,
-            lineNumbers: true,
-            autoCloseBrackets: true,
-            extraKeys: {
-                'Cmd-P': this.onAutoFormat,
-                'Ctrl-P': this.onAutoFormat
-            },
-            foldGutter: true,
-            scrollbarStyle: 'simple',
-            readOnly: disabled || readonly,
-            gutters,
-            lint: linter ? {getAnnotations: linter} : false,
-            autoFocus
-        };
-    }
-
-    onChange = ev => {
-        this.noteValueChange(ev.target.value);
-    };
+    manageCodeEditor = async (container: HTMLElement) => {
```

Comment:
Consider renaming to something like "createCodeEditor", since this method is doing the instantiation, and it seems clearer when providing it as a ref callback in the input cmp.